File: Number58_3.java
package algorithm;
import java.util.Scanner;
public class Number58_3 {

	public static void DFS(int n, int x) {
		if(x>n) return;
		else {
			DFS(n,x*2);
			DFS(n,x*2+1);
			System.out.print(x);
		}
	}

	public static void main(String[] args) {
		// 58_3. 이진트리 깊이우선트리(DFS)
		// 후휘순회 출력
		System.out.print("노드의 갯수를 입력하세요:");
		Scanner stdIn = new Scanner(System.in);
		int n = stdIn.nextInt();
		int x = n;
		DFS(n,1);
	}

}

File: number58_3.cpp
#include <stdio.h>
using namespace std;
int DFS(int n,int x){
	if(x>n){
		return 0;
	}
	else{
		DFS(n,2*x);
		DFS(n,2*x+1);
		printf("%d",x);
	}
}
int main() { 
// 58_3. 이진트리 깊이우선트리(DFS)
// 후위순회 출력.
	int n;
	printf("노드의 갯수를 입력하세요:");
	scanf("%d",&n);
	DFS(n,1);
	return 0;
 
}
